### ISSUE TITLE
Code changes don't trigger downstream targets

### ISSUE THREAD

**Issue by aammd** [richfitz/remake#30]
So maybe I've got the wrong idea about how this is supposed to work, but here's my situation. I'm building up a remakefile by writing functions (rules) as I go, checking my results and debugging with 

```r
m <- remake(envir=.GlobalEnv)
```

But when I decide to edit a function that is farther upstream, the downstream targets don't get remade. For example:

```yml
sources:
  - funs.R

targets:

  targ1:
    command: fun1()

  targ2:
    command: fun2(targ1)
```

In this example, `fun1()` and `fun2()` are defined in `funs.R`. I've found that if I run `remake()`, edit `fun1` and then rerun `remake()`, the target `targ1` is in the same state as it was before my edit to `fun1()`.  

From this, I'd infer that the functions and/or the scripts that are sourced to create them are not treated as dependencies of the targets they create. I worked around this with a fake target:

```yml
  code: 
    depends: funs.R
```

But maybe that is silly?

**Comment by richfitz** [comment]
Yeah that should work. Those are pretty fresh and there's some nasty problems with how the code dependency works that only appear when the environment that you're using is global.

**Comment by richfitz** [comment]
Could you send me a reproducible example of this, actually?  Contents of fun.R before and after change?  Unless this is just a sketch.  I might have something like this in the tests already but not tested with the active bindings enabled.

One of the reasons this is particularly sketchy at the moment is that I've tested things with the bindings in a separate environment rather than the global environment.

There's also some support for exporting things into another environment for testing and attaching that but  that doesn't work well with triggering rebuilds...

**Comment by richfitz** [comment]
This is closed now and I've added a test to catch it.

Be warned that there's at least two other nasty bugs lurking with the global environment approach: one is that the code dependency checking can trigger building of targets which becomes extremely slow.  I have a fix for that but it's at work.

The other is a mystery rebuild bug that I have yet to track down (I just haven't looked).  If you bump into that I'd love to get an reproducible example because the trigger case I saw it on takes ages to run.
